appserver/ejb/ejb-http-remoting/endpoint/src/main/java/fish/payara/ejb/invoke/InvokeEJBServlet.java

```diff
@@ -95,6 +109,14 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         JsonObject requestPayload = readJsonObject(request.getReader());
 
+        if(securityEnabled) {
+            for(String role : roles) {
+                if(!request.isUserInRole(role)) {
+                    response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+                }
+            }
+        }
```

Comment:
It likely wasn't necessary to add the feature in V0 endpoint. This is only kept for compatibility with client from 5.192, which didn't have the feature on client side.

The class definitely lacks a comment explaining that :)

Reply:
`InvokeEJBServlet` is already marked as `@Deprecated` but security feature is mandatory otherwise It will leave the security loophole as the Remote EJB can be accessible from V0 without checking the role.